addons/edi/models/models.py

```diff
@@ -15,14 +15,27 @@ def add_if_not_exists(cls):
     def wrapper(func):
         # pylint: disable=missing-docstring
         if hasattr(cls, func.__name__):
-            _logger.warning("%s.%s is already defined" % (cls.__name__, func.__name__))
+            _logger.info("%s.%s is already defined" % (cls.__name__, func.__name__))
```

Comment:
Why are we changing the log level here?

Reply:
If we ever have both `edi` and `udes_common` installed, on every startup/module upgrade of udes it spits out a bunch of warnings.

It's tricky, in that case we don't care for it, it's expected behaviour that the function already exists and we ignore.

But then in other cases we probably would want it to be raised as a warning instead. Would be good to get others input on this one

Reply:
Really we should address the TODO https://github.com/unipartdigital/udes-open/blob/story/606/addons/udes_common/models/models.py/#L13 but i haven't been able to think of a nice way of doing it

Reply:
mmmm, perhaps the model enhancements could be factored out into a separate addon?

Reply:
i'll have a think about it tomorrow and see if we can do anything

Reply:
Had a think about it last night, the way that makes the most sense to me is to just remove all of the duplicated functionality from `edi` which already exists in `udes_common` and just make `edi` depend on `udes_common`, and update the README in edi to reflect this dependency.
Then for edi, we still want the _valid_field_parameter override;
```python
from odoo.addons.udes_common.models import add_even_if_exists

@add_even_if_exists(models.BaseModel)
def _valid_field_parameter(self, field, name):
    ...
```
and any other specific functionality to edi

I'm not really sure why we care about edi being able to be installed without udes_common, they are both in public repos.
@PeterC10 @ws-mps @adamrpatrick you three might be able to give some counter-points to this

Reply:
https://github.com/unipartdigital/odoo-edi/commit/edb5589ea31a1055cdc1c24a7af7a272bd04bbeb
https://github.com/unipartdigital/udes-open/commit/45fba0fdf0db8af68df55316c1d4d9ba3c793421

^ I went ahead and tried out doing what i proposed, these two commits on story/606 branch on udes_open and odoo_edi are those changes.
Let me know if we want to go ahead with this or not next week.

Personally i think it's better this was as we don't have to maintain the same code in 2 places.. but it's up to you guys whether we are happy having the explicit module dependency in edi :+1:

Reply:
I was thinking we should put the shared code into a new addon that both this odoo-edi and udes_common could depend on.  However if we accept that our development of odoo-edi will diverge from upstream then depending on udes_common might be ok.  I don't think we would install this independently.

Reply:
We've discussed this a fair bit, for now let's change the log level to `debug`. That way it won't show up in a standard instance (as it is expected) but will be noted when running in debug.

I accept that it is a tricky one because in theory we should be able to run these repos independently, although like Kevin said we never would.

There is an argument to be had for putting the shared code into its own repo, but that is out of scope for now and not a priority.

Reply:
have changed to .debug in both udes_common and edi
also reverted above commits